src/js/api/auth/sign-up.mjs

```diff
@@ -0,0 +1,33 @@
+import { API_SOSIAL_URL } from "../constant-api.mjs";
+
+const action = "/auth/register";
+const method = "POST";
+
+/**
+ * This function creates a new user on use
+ * @param {string} profile The information used to create a new user
+ */
+async function signUp(profile) {
+  const signUpUrl = API_SOSIAL_URL + action;
+  const body = JSON.stringify(profile);
+
+  const response = await fetch(signUpUrl, {
+    method,
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body,
+  });
```

Comment:
```js
if (!response.ok) {
  throw new Error(response.status)
}
```

This will help you a lot with error handling

Reply:
Thank you very much, have implemented this in the project, should be better now.